warning: comparison of integer expressions of different signedness

While building 1.68.0 beta 1 on x86_64 linux cross-compiling to Windows 64 bits with mingw-w64 + g++ 8.2.0, I get this warning:
```
"x86_64-w64-mingw32-g++-8.2.0"   -I/softs/win64-mingw-8.2.0/release/iconv/include -I/softs/win64-mingw-8.2.0/release/gettext/include -I/softs/win64-mingw-8.2.0/release/bzip2/include -I/softs/win64-mingw-8.2.0/release/zlib/include -I/softs/win64-mingw-8.2.0/release/jpeg-turbo/include -I/softs/win64-mingw-8.2.0/release/xz/include -I/softs/win64-mingw-8.2.0/release/tiff/include -I/softs/win64-mingw-8.2.0/release/png/include -std=c++14 -O2 -DNDEBUG -m64 -mthreads -O3 -finline-functions -Wno-inline -Wall  -DBOOST_ALL_NO_LIB=1 -DBOOST_ASIO_NO_DEPRECATED -DBOOST_FILESYSTEM_NO_DEPRECATED -DBOOST_IOSTREAMS_USE_DEPRECATED -DBOOST_LOG_WITHOUT_EVENT_LOG -DNDEBUG  -I"." -I"/softs/win64-mingw-8.2.0/release/bzip2/include" -I"/softs/win64-mingw-8.2.0/release/zlib/include" -c -o "bin.v2/libs/iostreams/build/gcc-8.2.0/release/link-static/target-os-windows/threading-multi/mapped_file.o" "libs/iostreams/src/mapped_file.cpp"
libs/iostreams/src/mapped_file.cpp: In member function 'void boost::iostreams::detail::mapped_file_impl::try_map_file(boost::iostreams::detail::mapped_file_impl::param_type)':
libs/iostreams/src/mapped_file.cpp:324:29: warning: comparison of integer expressions of different signedness: 'boost::iostreams::stream_offset' {aka 'long long int'} and 'boost::iostreams::mapped_file_source::size_type' {aka 'long long unsigned int'} [-Wsign-compare]
             (SIZE_T) (size_ != max_length ? size_ : 0),
                       ~~~~~~^~~~~~~~~~~~~
```

--- a/src/mapped_file.cpp
+++ b/src/mapped_file.cpp
@@ -18,6 +18,7 @@
 #include <boost/iostreams/detail/system_failure.hpp>
 #include <boost/iostreams/device/mapped_file.hpp>
 #include <boost/throw_exception.hpp>
+#include <boost/numeric/conversion/cast.hpp>
 
 #ifdef BOOST_IOSTREAMS_WINDOWS
 # define WIN32_LEAN_AND_MEAN  // Exclude rarely-used stuff from Windows headers
@@ -323,7 +324,7 @@ void mapped_file_impl::try_map_file(param_type p)
             access,
             (DWORD) (p.offset >> 32),
             (DWORD) (p.offset & 0xffffffff),
-            (SIZE_T) (size_ != max_length ? size_ : 0), 
+            (SIZE_T) (numeric_cast<size_type>(size_) != max_length ? size_ : 0),
             (LPVOID) p.hint );
     if (!data)
         cleanup_and_throw("failed mapping view");

